### DISCUSSION TITLE
My Containers are scanned but not updated

### DISCUSSION THREAD

**Discussion by akib1689** [containrrr/watchtower#1735] [Q&A]
I Have down graded one of docker container version specifically to test this my current Dockerfile is like this:

```dockerfile
FROM containrrr/watchtower

# Set environment variables
ENV PUID=1000
ENV PGID=100
ENV WATCHTOWER_CLEANUP=true
ENV WATCHTOWER_LABEL_ENABLE=true
ENV WATCHTOWER_INCLUDE_RESTARTING=true
ENV WATCHTOWER_HTTP_API_TOKEN=mySecretToken

# Set the command to run Watchtower
CMD ["--interval", "86400", "--http-api-update", "--http-api-periodic-polls"]

# Expose port
EXPOSE 8080

# Mount the host's localtime file as read-only
VOLUME ["/etc/localtime:/etc/localtime:ro"]
```

and my run command looks like this:

```console
docker run -d     --name watchtower-container     -p 8080:8080     -v /var/run/docker.sock:/var/run/docker.sock     my-watchtower-image
```
and I have tagged my container perfectly. But this is the log of the watch tower when I manually trigger from the end point of watchtower

```console
time="2023-08-17T23:19:37Z" level=info msg="Watchtower 1.5.3"
time="2023-08-17T23:19:37Z" level=info msg="Using no notifications"
time="2023-08-17T23:19:37Z" level=info msg="Only checking containers using enable label"
time="2023-08-17T23:19:37Z" level=info msg="Scheduling first run: 2023-08-18 23:19:37 +0000 UTC"
time="2023-08-17T23:19:37Z" level=info msg="Note that the first check will be performed in 23 hours, 59 minutes, 59 seconds"
time="2023-08-17T23:19:37Z" level=info msg="The HTTP API is enabled at :8080."
time="2023-08-17T23:20:59Z" level=info msg="Updates triggered by HTTP API request."
time="2023-08-17T23:21:01Z" level=info msg="Session done" Failed=0 Scanned=1 Updated=0 notify=no
time="2023-08-17T23:28:17Z" level=info msg="Updates triggered by HTTP API request."
time="2023-08-17T23:28:20Z" level=info msg="Session done" Failed=0 Scanned=1 Updated=0 notify=no
```
May be I'm doing something wrong. My current Image version is 1.4.0 and the latest is 1.5.1
![image](https://github.com/containrrr/watchtower/assets/54747148/52681d2e-443a-4e86-a90c-6aa8a3a76eb8)
![image](https://github.com/containrrr/watchtower/assets/54747148/6f6382e4-a2f6-4a19-8039-646735792db1)

**Comment by akib1689** [comment]
okay I have run the command on debug mode The issue seems to be I am getting 401 unauthorized response for every request of docker hub. How to solve the issue?

I'm also attaching the docker debug response:

```console
time="2023-08-18T01:32:23Z" level=debug msg="Trying to load authentication credentials." container=/template-service image="docker.io/akib89/template-manager:v1.1.0"
time="2023-08-18T01:32:23Z" level=debug msg="No credentials for docker.io found" config_file=/config.json
time="2023-08-18T01:32:23Z" level=debug msg="Got image name: docker.io/akib89/template-manager:v1.1.0"
time="2023-08-18T01:32:23Z" level=debug msg="Checking if pull is needed" container=/template-service image="docker.io/akib89/template-manager:v1.1.0"
time="2023-08-18T01:32:23Z" level=debug msg="Building challenge URL" URL="https://index.docker.io/v2/"
time="2023-08-18T01:32:24Z" level=debug msg="Got response to challenge request" header="Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\"" status="401 Unauthorized"
time="2023-08-18T01:32:24Z" level=debug msg="Checking challenge header content" realm="https://auth.docker.io/token" service=registry.docker.io
time="2023-08-18T01:32:24Z" level=debug msg="Setting scope for auth token" image=docker.io/akib89/template-manager scope="repository:akib89/template-manager:pull"
time="2023-08-18T01:32:24Z" level=debug msg="No credentials found."
time="2023-08-18T01:32:24Z" level=debug msg="Parsing image ref" host=index.docker.io image=docker.io/akib89/template-manager normalized="docker.io/akib89/template-manager:v1.1.0" tag=v1.1.0
time="2023-08-18T01:32:24Z" level=debug msg="Doing a HEAD request to fetch a digest" url="https://index.docker.io/v2/akib89/template-manager/manifests/v1.1.0"
time="2023-08-18T01:32:25Z" level=debug msg="Found a remote digest to compare with" remote="sha256:dd533eddb54bdb1bc04d346fcd29d89d75f8afdb52d0d1692dac1afa73747169"
time="2023-08-18T01:32:25Z" level=debug msg=Comparing local="sha256:dd533eddb54bdb1bc04d346fcd29d89d75f8afdb52d0d1692dac1afa73747169" remote="sha256:dd533eddb54bdb1bc04d346fcd29d89d75f8afdb52d0d1692dac1afa73747169"
time="2023-08-18T01:32:25Z" level=debug msg="Found a match"
time="2023-08-18T01:32:25Z" level=debug msg="No pull needed. Skipping image."
time="2023-08-18T01:32:25Z" level=debug msg="No new images found for /template-service"
```

**Reply by piksel** [reply]
No, it works correctly, but the remote hash is the same as the local one, so no update is available.

```
Comparing local="sha256:dd533eddb54bdb1bc04d346fcd29d89d75f8afdb52d0d1692dac1afa73747169" remote="sha256:dd533eddb54bdb1bc04d346fcd29d89d75f8afdb52d0d1692dac1afa73747169"
Found a match
No pull needed. Skipping image.
No new images found for /template-service
```

**Comment by piksel** [comment]
> I Have down graded one of docker container version

How did you do this? Watchtower reports that it checked one container for an updated image, but found no newer version. 

If you use the version tags in the screenshot above it won't work, since watchtower only checks images with the same tag. If you want "latest minor' updates, you need to tag your images with both `v1` **and** `v1.5.1` (and if you want "latest patch", you would also need `v1.5`).

Then you would start your container with `akib-89/view-service:v1` or `:v1.5` etc.

**Reply by akib1689** [reply]
I'm using v1.4.0 and there is a version with tag v1.5.1 should not it update that? I'm not understanding what am I doing wrong. and I manually tested downgraded version simply by running container with lower version image

**Reply by piksel** [reply]
No, Watchtower will only update to a newer image _with the same tag_. 

The tags can be any string of text, not just version numbers, so there is no way for Watchtower to know what tags should be considered updates. 

So, to support your scenario you should also tag the images with `v1`. That way, when you create a new version, `v1` will point to `v1.5.1` or whatever the new version is, and watchtower will update to that version.

**Comment by akib1689** [comment]
I think I'm understanding something If the image with tag **v1.4.0** is updated in the remote then the watchtower will work and since my version is changed It is not pulling the image. Is there any way to use like this behavior? i.e. It will check based on the version number present in the repository?
